nilearn/maskers/surface_labels_masker.py

```diff
@@ -0,0 +1,463 @@
+"""Extract data from a SurfaceImage, averaging over atlas regions."""
+
+import warnings
+
+import numpy as np
+from joblib import Memory
+from sklearn.base import BaseEstimator, TransformerMixin
+
+from nilearn import signal
+from nilearn._utils import _constrained_layout_kwargs, fill_doc
+from nilearn._utils.cache_mixin import CacheMixin, cache
+from nilearn._utils.class_inspect import get_params
+from nilearn._utils.helpers import is_matplotlib_installed
+from nilearn.maskers._utils import (
+    check_same_n_vertices,
+    compute_mean_surface_image,
+    get_min_max_surface_image,
+)
+from nilearn.surface import SurfaceImage
+
+
+@fill_doc
+class SurfaceLabelsMasker(TransformerMixin, CacheMixin, BaseEstimator):
+    """Extract data from a SurfaceImage, averaging over atlas regions.
+
+    Parameters
+    ----------
+    labels_img : :obj:`~nilearn.surface.SurfaceImage` object
+        Region definitions, as one image of labels.
+
+    labels : :obj:`list` of :obj:`str`, default=None
+        Full labels corresponding to the labels image.
+        This is used to improve reporting quality if provided.
+
+        .. warning::
+            The labels must be consistent with the label values
+            provided through ``labels_img``.
+
+    background_label : :obj:`int` or :obj:`float`, default=0
+        Label used in labels_img to represent background.
+
+        .. warning::
+
+            This value must be consistent with label values
+            and image provided.
```

Comment:
Not 100% sure it makes sense to have a background label on a **surface** label masker, but in case people project a volume atlas onto a surface, it may happen to have some vertices that are 'unassinged to a region' so probably better to keep this.

May also happen if the atlas definition does not cover all the brain regions (for example only map visual areas).